easybuild/easyblocks/x/xalt.py

```diff
@@ -0,0 +1,163 @@
+            # and the CUDA module is loaded
+            cuda_root = get_software_root('CUDA')
+            if cuda_root:
+                nvml_h = os.path.join(cuda_root, "include", "nvml.h")
+                if os.path.isfile(nvml_h):
+                    self.cfg.update('configopts', '--with-trackGPU=yes')
+                    self.cfg['gpu_tracking'] = True
+        else:
+            # User disabled
+            self.cfg.update('configopts', '--with-trackGPU=no')
+
+        # MySQL
+        if self.cfg['mysql'] is True:
+            self.cfg.update('configopts', '--with-MySQL=yes')
+        else:
+            self.cfg.update('configopts', '--with-MySQL=no')
+
+        # Configure
+        super(EB_XALT, self).configure_step()
+
+    def make_module_extra(self, *args, **kwargs):
+        txt = super(EB_XALT, self).make_module_extra(*args, **kwargs)
+
+        txt += self.module_generator.prepend_paths('LD_PRELOAD', 'lib64/libxalt_init.%s' % get_shared_lib_ext())
+        txt += self.module_generator.set_environment('XALT_DIR', self.installdir)
+        txt += self.module_generator.set_environment('XALT_ETC_DIR', '%s' % os.path.join(self.installdir, 'etc'))
+        txt += self.module_generator.set_environment('XALT_EXECUTABLE_TRACKING',
+                                                     ('no', 'yes')[bool(self.cfg['executable_tracking'])])
+        txt += self.module_generator.set_environment('XALT_GPU_TRACKING',
+                                                     ('no', 'yes')[bool(self.cfg['gpu_tracking'])])
+        if self.cfg['transmission'].lower() == 'curl' and self.cfg['logging_url']:
+            txt += self.module_generator.set_environment('XALT_LOGGING_URL', self.cfg['logging_url'])
+        txt += self.module_generator.set_environment('XALT_SCALAR_SAMPLING',
+                                                     ('no', 'yes')[bool(self.cfg['scalar_sampling'])])
+
+        # In order to track containerized executables, bind mount the XALT
+        # directory in the Singularity container and preload the XALT library
+        # https://xalt.readthedocs.io/en/latest/050_install_and_test.html#xalt-modulefile
+        txt += self.module_generator.prepend_paths('SINGULARITY_BINDPATH', '')
+        txt += self.module_generator.prepend_paths('SINGULARITYENV_LD_PRELOAD',
+                                                   'lib64/libxalt_init.%s' % get_shared_lib_ext())
+
+        return txt
+
+    def make_module_req_guess(self):
+        """Custom guesses for environment variables"""
+        return {'COMPILER_PATH': 'bin',
+                'PATH': 'bin'}
+
+    def sanity_check_step(self):
+        """Custom sanity check"""
+        custom_paths = {
+            'files': ['bin/ld', 'bin/ld.gold', 'bin/xalt_extract_record',
+                      'lib64/libxalt_init.%s' % get_shared_lib_ext()],
+            'dirs': ['bin', 'libexec', 'sbin'],
+        }
+
+        super(EB_XALT, self).sanity_check_step(custom_paths=custom_paths)
```

Comment:
We should also run `xalt_configuration_report` as a sanity check command, see also https://github.com/easybuilders/easybuild-easyconfigs/pull/9792#issuecomment-631248656